nucypher/utilities/porter/control/interfaces.py

```diff
@@ -31,23 +34,26 @@ def __init__(self, porter: 'Porter' = None, *args, **kwargs):
     def get_ursulas(self,
                     quantity: int,
                     duration_periods: int,
-                    exclude_ursulas: List[str],
-                    include_ursulas: List[str]) -> dict:
-        # Steps (analogous to nucypher.character.control.interfaces):
-        # 1. creation of relevant objects / setup
-        # 2. call self.implementer.some_function() i.e. Porter learner has an associated function to call
-        # 3. create response
-        pass
+                    exclude_ursulas: Optional[List[ChecksumAddress]] = None,
+                    include_ursulas: Optional[List[ChecksumAddress]] = None) -> dict:
+        ursulas_info = self.implementer.get_ursulas(quantity=quantity,
+                                                    duration_periods=duration_periods,
+                                                    exclude_ursulas=exclude_ursulas,
+                                                    include_ursulas=include_ursulas)
+
+        response_data = {
+            "ursulas": ursulas_info
+        }
+        return response_data
 
     @attach_schema(porter_schema.AlicePublishTreasureMap)
     def publish_treasure_map(self,
                              treasure_map: bytes,
                              bob_encrypting_key: bytes) -> dict:
-        # Steps (analogous to nucypher.character.control.interfaces):
-        # 1. creation of relevant objects / setup
-        # 2. call self.implementer.some_function() i.e. Porter learner has an associated function to call
-        # 3. create response
-        pass
+        bob_enc_key = UmbralPublicKey.from_bytes(bob_encrypting_key)
+        self.implementer.publish_treasure_map(treasure_map_bytes=treasure_map,
+                                              bob_encrypting_key=bob_enc_key)
+        return {}
```

Comment:
I'm wondering what can we return here, like the list of Ursulas holding the TMap.

Reply:
I guess my question would be what's the advantage there? The distribution of the treasure map is based on the hamming distance algorithm based on bob's encrypting key. Bob will inherently use that same algorithm through Porter via `/get_treasure_map`.